### PR TITLE
Including param validation for SCE pilot

### PR DESCRIPTION
**Pull request by claalmve**
# Description of PR

**Summary of changes and related issue**
Including parameter validation for SCE pilot. This should make the code more readable. Additionally, `cava_data` now catches multiple logic errors at once, allowing the user to fix multiple issues at once rather than fixing them one by one. 

WIP: Catching multiple input validation errors at once. I.e. inputting `meannn` for `metric_calc` and `Dynamicallll` for `downscaling_method` will only show the `meannn` as an issue. Once that is addressed, then it shows `Dynamicallll` as an issue.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: climakitae/explore/vulnerability.py [hunk 1]
```diff
@@ -573,151 +644,112 @@ def cava_data(
...
-        if len(loaded_data) == 1:
-            loaded_data = loaded_data[0]
+            if export_method == "calculate":
+                return {"calc_data": metric_data}
+            elif export_method == "both":
+                return {"calc_data": metric_data, "raw_data": loaded_data}
 
-        if export_method == "calculate":
+        elif export_method == "None":  # Specific for table generation
             return {"calc_data": metric_data}
```
Comment (vicford) [inline climakitae/explore/vulnerability.py:755]:
```suggestion
        elif export_method == "None":  # Specific for table generation
            print('Data export selection set to "None"; no data is exported!")
            return {"calc_data": metric_data}
```

**Review by vicford — APPROVED** [review]
@claalmve this looks fine to me! But I would also recommend another developer check this for more tech-y review. I tested a bunch of bad options and really appreciate how clean the warning message is with the valid options. I did add one print statement for the export = None option just for clarity

**Comment by Tianchi-Liu** [comment]
I am not aware of any usage of `param` that’s more appropriate here. Looks like `CavaParams` helps to capture inputs to `cava_data` that are not intended to be supported. The introduction of `param` makes input validation much more efficient👍

**Comment by Tianchi-Liu** [comment]
It’d be great if users are asked to clarify their intention if they set `approach="time"` but also provided a warming level.

**Pull request merged**